gui/options-dialog.cpp

```diff
@@ -163,3 +169,23 @@ void OptionsDialog::doCancel()
     close();
 }
 
+void OptionsDialog::browse_datalogging_file()
+{
+    QString filename = ui.tracklogging_filenameedit->text();
+    if (filename.isEmpty())
+        filename = QDir::currentPath();
+    /* Sometimes this function freezes the app before opening the dialog.
```

Comment:
Remove usages of `CoInitializeEx` from the tree and it should be fine. If this works I'll fix it in the UI thread.